src/components/mgt-login/mgt-login.ts

```diff
@@ -239,6 +188,94 @@ export class MgtLogin extends MgtBaseComponent {
     this._loading = false;
   }
 
+  /**
+   * Render the button.
+   *
+   * @protected
+   * @memberof MgtLogin
+   */
+  protected renderButton() {
+    return html`
+      <button ?disabled="${this._loading}" class="login-button" role="button">
+        ${this.renderButtonContent()}
+      </button>
+    `;
+  }
+
+  /**
+   * Render the details flyout.
+   *
+   * @protected
+   * @memberof MgtLogin
+   */
+  protected renderFlyout() {
+    return html`
+      <mgt-flyout .isOpen=${this._showMenu}>
+        ${this.renderFlyoutContent()}
+      </mgt-flyout>
+    `;
+  }
+
+  /**
+   * Render the flyout menu content.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtLogin
+   */
+  protected renderFlyoutContent() {
+    if (!this.userDetails) {
+      return;
+    }
+
+    return html`
+      <div slot="flyout" class="flyout">
+        <div class="popup">
+          <div class="popup-content">
+            <div>
+              <mgt-person .personDetails=${this.userDetails} .personImage=${this._image} show-name show-email />
+            </div>
+            <div class="popup-commands">
+              <ul>
+                <li>
+                  <button class="popup-command" @click=${this.logout} aria-label="Sign Out">
+                    Sign Out
+                  </button>
+                </li>
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * Render the button content.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtLogin
+   */
+  protected renderButtonContent() {
+    if (this.userDetails) {
+      return html`
+        <mgt-person .personDetails=${this.userDetails} .personImage=${this._image} show-name />
+      `;
+    } else {
+      return html`
+        <i class="login-icon ms-Icon ms-Icon--Contact"></i>
+        <span aria-label="Sign In">
+          Sign In
+        </span>
+      `;
+    }
+  }
+
+  private handleWindowClick(e: MouseEvent) {
+    this._showMenu = false;
+  }
+
   private onClick(event: MouseEvent) {
```

Comment:
Just a thought, but should we consider making this (or a differently named method) public to avoid having the click handler outside of the button?

Reply:
That's a good point. We also don't expose a good way to show/hide the menu flyout, which can make it difficult to override further. Although I don't necessarily like the idea of exposing `onClick`.

I've made an adjustment to make it possible for the developer to reproduce the same effect as the click handler, if they happen to override the `renderButton` function. That's really the only time that the click handler is relevant. If the developer has full control of the Button rendering, then they can wire up their own click handler, and use protected functions to manipulate the flyout (which can be overridden as well!):
- `showFlyout()`
- `hideFlyout()`
- `toggleFlyout()`

I've simplified the `onClick()` to look like this, which shouldn't be a burden for the developer to reproduce in their own click handler.
```
private onClick(event: MouseEvent) {
  if (this.userDetails) {
    this.toggleFlyout();
  } else {
    this.login();
  }
}
```
 Thoughts?

Reply:
I love it